pkg/platform/model/inventory.go

```diff
@@ -679,3 +679,97 @@ func FilterCurrentPlatform(hostPlatform string, platforms []strfmt.UUID, preferr
 
 	return platformIDs[0], nil
 }
+
+// CreateNewIngredientVersionRevision creates a new revision for the provided ingredient.
+// The new revision only updates the comment and dependencies of the ingredient version, all further fields are kept as-is.
+func CreateNewIngredientVersionRevision(ingredient *inventory_models.FullIngredientVersion, comment string, dependencies []inventory_models.Dependency, auth *authentication.Auth) (*inventory_operations.AddIngredientVersionRevisionOK, error) {
+	// Retrieve its latest revision to copy all data - but comment and dependencies - from
+	getParams := inventory_operations.NewGetIngredientVersionRevisionsParams()
+	getParams.SetIngredientID(*ingredient.IngredientID)
+	getParams.SetIngredientVersionID(*ingredient.IngredientVersionID)
+
+	client := inventory.Get(auth)
+	revisions, err := client.GetIngredientVersionRevisions(getParams, auth.ClientAuth())
+	if err != nil {
+		return nil, errs.Wrap(err, "error getting version revisions")
+	}
+	revision := revisions.Payload.IngredientVersionRevisions[len(revisions.Payload.IngredientVersionRevisions)-1]
+
+	// Prepare new ingredient version revision params to create a new revision
+	// This leaves all the attributes untouched, but dependencies and comments
+	newParams := inventory_operations.NewAddIngredientVersionRevisionParams()
+	newParams.SetIngredientID(*ingredient.IngredientID)
+	newParams.SetIngredientVersionID(*ingredient.IngredientVersionID)
+
+	// Extract build script IDs
+	var buildScriptIDs []strfmt.UUID
+	for _, script := range revision.BuildScripts {
+		buildScriptIDs = append(buildScriptIDs, *script.BuildScriptID)
+	}
+
+	// Replicate patches
+	var patches []*inventory_models.IngredientVersionRevisionCreatePatch
+	for _, patch := range revision.Patches {
+		patches = append(patches, &inventory_models.IngredientVersionRevisionCreatePatch{
+			PatchID:        patch.PatchID,
+			SequenceNumber: patch.SequenceNumber,
+		})
+	}
+
+	// Retrieve and prepare default and override option sets
+	optsetParams := inventory_operations.NewGetIngredientVersionIngredientOptionSetsParams()
+	optsetParams.SetIngredientID(*ingredient.IngredientID)
+	optsetParams.SetIngredientVersionID(*ingredient.IngredientVersionID)
+
+	response, err := client.GetIngredientVersionIngredientOptionSets(optsetParams, auth.ClientAuth())
+	if err != nil {
+		return nil, errs.Wrap(err, "error getting optsets")
+	}
+
+	var default_optsets []strfmt.UUID
+	var override_optsets []strfmt.UUID
+	for _, optset := range response.Payload.IngredientOptionSetsWithUsageType {
+		switch *optset.UsageType {
+		case "default":
+			default_optsets = append(default_optsets, *optset.IngredientOptionSetID)
+		case "override":
+			override_optsets = append(override_optsets, *optset.IngredientOptionSetID)
```

Comment:
Suggest adding a default handler that errors out (eg. unrecognized type). Even if there are only 2, that should future proof us somewhat in the sense that if this change in the future it'll be immediately apparent that our code needs updating.